File: Data Structures & Algorithms/longest-palindromic-substring/submission-3.py
class Solution:
    def longestPalindrome(self, s: str) -> str:
        
        ret = ""
        for i in range(len(s)):
            left = i
            right = i

            while(left >= 0 and right < len(s) and s[left] == s[right]):
                if(len(ret) < right - left + 1):
                    ret = s[left: right + 1]
                left -= 1
                right += 1
            
            left = i 
            right = i + 1
            while(left >= 0 and right < len(s) and s[left] == s[right]):
                if(len(ret) < right - left + 1):
                    ret = s[left: right + 1]
                left -= 1
                right += 1

        return ret

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-3.java
class Solution {
    public int maxProfit(int[] prices) {
        if(prices.length == 1){
            return 0;
        }
        int maxVal = 0;
        int left = 0;
        int right = 1;
        while(right < prices.length){
            if(prices[left] <= prices[right]){
                maxVal = Math.max(maxVal, prices[right] - prices[left]);
            }else{
                left = right;
            }
            right++;
        }
        return maxVal;
    }
}
